src/third/redux-persist/createPersistor.js

```diff
@@ -111,7 +118,7 @@ export default function createPersistor (store, config) {
     }
 
     // Record success.
-    lastWrittenState = state
+    outstandingKeys.length = 0;
```

Comment:
> Interesting, I guess this empties out the array without leaving anything behind.

Yep -- looks kind of funny, but it's how you spell `clear()` on a JS `Array`.


> Since we're not using `outstandingKeys = []`, should we make it a `const` variable instead of `let`?

I don't understand the first half of this. But yeah, I agree: we never alter this binding (just mutate the object itself), so it should be a `const`. Thanks for pointing that out.

If we had ESLint running on this file, it'd have spotted and fixed that automatically :slightly_smiling_face:

Reply:
> I don't understand the first half of this.

`outstandingKeys = []` would be pretty much a fine substitute for `outstandingKeys.length = 0`, wouldn't it? (If we kept the `let`, anyway.)

It's what I would have reached for, I think. But I didn't really know about setting `length` to 0, and that may be better.

Reply:
(This is moot with a Set anyway.)

Reply:
> `outstandingKeys = []` would be pretty much a fine substitute for `outstandingKeys.length = 0`, wouldn't it? (If we kept the `let`, anyway.)

Ah, got it.

Yeah, that would have worked here too (modulo the issue that it should be a set instead of an array.) Similarly, in the merged version where it's a set, instead of `outstandingKeys.clear()` we could say `outstandingKeys = new Set()`.

Saying `.clear()` or `.length = 0` is a bit cleaner because it's parallel to the other place we update the data structure. We're saying `.push(key)` (in the array version), not `= [...outstandingKeys, key]`, so we're mutating the data structure itself, not mutating the binding of the name `outstandingKeys` to point to a new data structure. With `.length = 0`, we again mutate the data structure itself, rather than mutating the binding to point to a new one.

It's also slightly more efficient: we save the work of allocating (and later GC'ing) a new data structure. Though that's not really significant here, as one small array or set is not much compared to the work of serializing big chunks of our state tree and writing them out.

Reply:
Makes sense. Thanks! :)